### PR TITLE
:test_tube: added unit test for auth0 provider

### PR DESCRIPTION
**Pull request by Takayuki-Y5991**
## Summary by Sourcery

テスト:
- Auth0プロバイダーのユニットテストを追加。以下のようなさまざまなシナリオをカバーしています：
  - 認証URLの生成
  - ユーザー情報の取得
  - トークンのリフレッシュ
  - 多要素認証（MFA）操作
  - コンテキストのキャンセル
  - エッジケース
  - トークン検証
  - 認証コードの交換

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Tests:
- Added unit tests for the Auth0 provider, covering various scenarios like generating authorization URLs, retrieving user information, refreshing tokens, MFA operations, context cancellation, edge cases, token verification, and exchanging authorization codes.

</details>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,15 +38,29 @@ tools:
...
\ No newline at end of file
+	go run ./cmd/main.go
+
+# Run verbose tests
+test-verbose:
+	gotestsum --format standard-verbose
+
+# Run tests with JSON output
+test-json:
+	gotestsum --format testname --jsonfile test-output.json
```
[the remark this answers is not in the document]
Comment (Takayuki-Y5991) [reply Makefile:61]: @sourcery-ai
Please push up issue
